packages/flutter_tools/lib/src/commands/build_aot.dart

```diff
@@ -33,11 +33,16 @@ class BuildAotCommand extends BuildSubCommand {
         hide: !verboseHelp,
         help: 'Preview Dart 2.0 functionality.',
       )
+      ..addFlag('build-shared-library',
+        negatable: false,
+        defaultsTo: false,
+        help: 'Compile to a *.so file (Android only; requires NDK).'
```

Comment:
maybe make it work on both iOS and Android - it will be a NOP on iOS. 

You can then change this comment to `" (Requires NDK when building for Android)"`

Reply:
Done.